README.md
### DirList

Page module for the the Content Management Systems [WebsiteBaker] [1] or [LEPTON CMS] [2] to show the content of a /MEDIA directory at any page in the frontend. 

IMPORTANT: This addon is DEPRECATED and will be no longer supported. Please use the more powerfull [kitDirList] [6] instead!

#### Requirements

* minimum PHP 4.x
* using [WebsiteBaker] [1] _or_ using [LEPTON CMS] [2]

#### Installation

* download the actual [DirList_x.xx.zip] [3] installation archive
* in CMS backend select the file from "Add-ons" -> "Modules" -> "Install module"

#### First Steps

Please visit the [phpManufaktur] [4] to get more informations about **DirList** and join the [Addons Support Group] [5].

[1]: http://websitebaker2.org "WebsiteBaker Content Management System"
[2]: http://lepton-cms.org "LEPTON CMS"
[3]: https://addons.phpmanufaktur.de/download.php?file=DirList
[4]: http://phpmanufaktur.de/cms/topics/dirlist.php
[5]: https://phpmanufaktur.de/support
[6]: https://addons.phpmanufaktur.de/download.php?file=kitDirList


CHANGELOG
### DirList
2007-2012 by phpManufaktur, Berlin (Germany)
ralf.hertsch@phpmanufaktur.de
http://phpmanufaktur.de


Release 0.23 - 2012-06-17

* moved repository from SVN to GitHub
* updated all header informations
* changed license from GNU GPL to MIT
* added LICENSE
* added CHANGELOG
* added README.md

Release 0.22 - 2010-07-29

* added french language file FR.php - many thanks to EmGel

Release 0.21 - 2008-05-14

* fixed problem reading WB settings from database

Release 0.20 - 2008-04-12

* fixed use of specific PHP5 features prohibited proper execution under PHP4

Release 0.19 - 2008-04-04

* fixed class.mimetypes.php returns wrong icon for application/msword
* fixed problem getting Media Directories with apache at windows pc
  (many thanks to Robert Schreiner for solution)
* fixed empty $_REQUEST's causes error messages in backend

Release 0.18 - 2008-02-18

* fixed problem if class.parser.php is used in different modules

Release 0.17 - 2007-11-14

* added Italian language file IT.php - many thanks to Antonello Alonzi

Release 0.16 - 2007-11-08

* fixed some mispellings...
* added switch to open files in a new window (target="_blank")

Release 0.15a - 2007-11-03

* updated Dutch language file NL.php - many thanks to Marco Loehnen
 
Release 0.15 - 2007-10-31

* fixed DirList shows hidden files
* changed structure of language files
* added filter for including and excluding files in DirList

Release 0.14 - 2007-10-18

* fixed HTML/XML - some markup language errors prevented DirList from passing
  W3C Markup Language Validation
* Bug in WB 2.6.7 impede modules from passing W3C Validation, please check
  Ticket #521 --> http://projects.websitebaker.org/websitebaker2/ticket/521
  for further informations and fix-up
 
 Release 0.13 - 2007-10-14

* changed replaced variable $module_directory with hardcoded 'dirlist'
* added Dutch language file NL.php - many thanks to Marco Loehnen

Release 0.12 - 2007-10-13

* fixed show multiple DirList's within a single page
 
Release 0.11 - 2007-10-13

* added frontend shows 2 dots instead of filename if directory is empty
* fixed backend lost values in case of errors while saving settings.
* added click on table header will sort DirList by filename ascending or 
  descending, size ascending or descending, date ascending or descending
* added show mimetype icons in DirList
* added upgrade.php for automated upgrades

Release 0.10 - 2007-10-07

* initial release


upgrade.php
<?php

/**
 * DirList
 *
 * @author Ralf Hertsch <ralf.hertsch@phpmanufaktur.de>
 * @link http://phpmanufaktur.de
 * @copyright 2007 - 2012
 * @license MIT License (MIT) http://www.opensource.org/licenses/MIT
 */

// include class.secure.php to protect this file and the whole CMS!
if (defined('WB_PATH')) {
  if (defined('LEPTON_VERSION'))
    include(WB_PATH.'/framework/class.secure.php');
}
else {
  $oneback = "../";
  $root = $oneback;
  $level = 1;
  while (($level < 10) && (!file_exists($root.'/framework/class.secure.php'))) {
    $root .= $oneback;
    $level += 1;
  }
  if (file_exists($root.'/framework/class.secure.php')) {
    include($root.'/framework/class.secure.php');
  }
  else {
    trigger_error(sprintf("[ <b>%s</b> ] Can't include class.secure.php!", $_SERVER['SCRIPT_NAME']), E_USER_ERROR);
  }
}
// end include class.secure.php

// manually include the config.php file (defines the required constants)
require('../../config.php');

// Modul Informationen
require_once('info.php');

if(!file_exists(WB_PATH .'/modules/dirlist/languages/' .LANGUAGE .'.php')) {
	require_once(WB_PATH .'/modules/dirlist/languages/DE.php');
} else {
		require_once(WB_PATH .'/modules/dirlist/languages/' .LANGUAGE .'.php');
}


/**
 * Upgrade from 0.10 to 0.11
 *
 * --> add row "sort" for options sorting DirList
 *
 */

$prompt = '';
$upgradeError = false;

$thisQuery = "DESCRIBE ".TABLE_PREFIX."mod_dirlist";
$oldErrorReporting = error_reporting(0);
$sql_result = $database->query($thisQuery);
error_reporting($oldErrorReporting);
if ($database->is_error()) {
	// Fehlermeldung anzeigen
	$upgradeError = true;
	$prompt .= sprintf(dl_error_describe_table,$database->get_error()); }
else {
	$fields = array();
	$searchField = 'sort';
	while (($data = $sql_result->fetchRow())) {
		$fields[] = $data["Field"];	}
	if (in_array($searchField, $fields)) {
		// Spalte bereits vorhanden
		$prompt .= sprintf(dl_upgrade_field_exists,'sort');	}
	else {
		// Tabelle muss ergaenzt werden
		$thisQuery = "ALTER TABLE ".TABLE_PREFIX."mod_dirlist ADD sort INT(11) DEFAULT 1";
		$oldErrorReporting = error_reporting(0);
		$sql_result = $database->query($thisQuery);
		error_reporting($oldErrorReporting);
		if ($database->is_error()) {
			// Fehler beim Einfuegen des Feldes
			$upgradeError = true;
			$prompt .= sprintf(dl_error_insert_field,'sort',$database->get_error()); }
	  else {
	  	// Feld erfolgreich eingefuegt
	  	$prompt .= sprintf(dl_upgrade_insert_field,'sort'); }
	}
}

/**
 * Upgrade from 0.14 to 0.15
 *
 * --> Option for include or exclude Files by Extension
 *
 */
$thisQuery = "DESCRIBE ".TABLE_PREFIX."mod_dirlist";
$oldErrorReporting = error_reporting(0);
$sql_result = $database->query($thisQuery);
error_reporting($oldErrorReporting);
if ($database->is_error()) {
  $upgradeError = true;
	$prompt .= sprintf(dl_error_describe_table, $database->get_error()); }
else {
	$fields = array();
	while (($data = $sql_result->fetchRow())) {
	  $fields[] = $data["Field"]; }
	if ((in_array('exclude', $fields)) && (in_array('extensions', $fields))) {
	  // Die Spalten sind bereits vorhanden
	  $prompt .= sprintf(dl_upgrade_field_exists,'exclude');
	  $prompt .= sprintf(dl_upgrade_field_exists,'extensions');	}
	else {
	  // Tabelle muss ergaenzt werden
		$oldErrorReporting = error_reporting(0);
		$thisQuery = "ALTER TABLE ".TABLE_PREFIX."mod_dirlist ADD exclude INT(11) DEFAULT 1";
		$database->query($thisQuery);
		if ($database->is_error()) {
		  $upgradeError = true;
		  $prompt .= sprintf(dl_error_insert_field,'exlude',$database->get_error()); }
    else {
			$prompt .= sprintf(dl_upgrade_insert_field,'exclude');
	    $thisQuery = "ALTER TABLE ".TABLE_PREFIX."mod_dirlist ADD extensions VARCHAR(255) DEFAULT ''";
		  $database->query($thisQuery);
		  if ($database->is_error()) {
		    $upgradeError = true;
		    $prompt .= sprintf(dl_error_insert_field,'extensions',$database->get_error()); }
		  else {
	  	  // Feld erfolgreich eingefuegt
	  	  $prompt .= sprintf(dl_upgrade_insert_field,'extensions'); }
    }
	  error_reporting($oldErrorReporting);
	}
}

/**
 * Upgrade from 0.15 to 0.16
 *
 * --> Option for show files in a new window
 *
 */
$thisQuery = "DESCRIBE ".TABLE_PREFIX."mod_dirlist";
$oldErrorReporting = error_reporting(0);
$sql_result = $database->query($thisQuery);
error_reporting($oldErrorReporting);
if ($database->is_error()) {
  $upgradeError = true;
	$prompt .= sprintf(dl_error_describe_table, $database->get_error()); }
else {
	$fields = array();
	while (($data = $sql_result->fetchRow())) {
	  $fields[] = $data["Field"]; }
	if (in_array('blank', $fields)) {
	  // Die Spalten sind bereits vorhanden
	  $prompt .= sprintf(dl_upgrade_field_exists,'blank');	}
	else {
	  // Tebelle muss ergaenzt werden
		$thisQuery = "ALTER TABLE ".TABLE_PREFIX."mod_dirlist ADD blank INT(11) DEFAULT 0";
		$oldErrorReporting = error_reporting(0);
		$sql_result = $database->query($thisQuery);
		error_reporting($oldErrorReporting);
		if ($database->is_error()) {
			// Fehler beim Einfuegen des Feldes
			$upgradeError = true;
			$prompt .= sprintf(dl_error_insert_field,'blank',$database->get_error()); }
	  else {
	  	// Feld erfolgreich eingefuegt
	  	$prompt .= sprintf(dl_upgrade_insert_field,'blank'); }
	}
}


// Ergebnis des Upgrade melden
if ($upgradeError) {
	$admin->print_error($prompt); }
else {
	$admin->print_success($prompt); }


?>